### PR TITLE
Document howto publish branch ubuntu-ppa for Launchpad builder

### PR DESCRIPTION
**Pull request by ottok**
Related: #55

### CODE REVIEW

**Comment by ericzolf** [comment]
Just for planning, is this something that can be done post-2.0.0 or do you want it in 2.0.0?

**Comment by ottok** [comment]
This can be post 2.0.0.

Runner requires SSH access in a way I have not yet been able to figure out
how to do safely.

**Comment by ericzolf** [comment]
Did you see https://docs.travis-ci.com/user/deployment/launchpad/ ?

**Comment by ottok** [comment]
Status is still that runner requires SSH access in a way I have not yet been able to figure out
how to do safely.

[Triggering Launchpad builds](https://docs.travis-ci.com/user/deployment/launchpad/) already happens automatically. The issue is that the source needs to have the version updated and commited to git before the Launchpad build is triggered, since we maintain dynamic version on SCM_VERSION and Launchpad (or any other real Debian environment) does not allow packages to change their own versions on the fly.

**Comment by ottok** [comment]
In the mean time, these are the steps to publish latest git master on the PPA:
```
git checkout master; git pull # Ensure your local master is up-to-date with origin master
git checkout bugfix/deb-version-and-ppa # Checkout this branch
make dist_deb # Runs until it fails
git push --force --set-upstream origin ubuntu-ppa # Run this manually with your SSH keys in place
```

Then wait until source code sync runs or trigger it manually at https://code.launchpad.net/~rdiff-backup/rdiff-backup/+git/trunk

Then wait for the daily build to run or trigger it manually at https://code.launchpad.net/~rdiff-backup/+recipe/rdiff-backup-git-master

Full build logs will be visible at https://launchpad.net/~rdiff-backup/+archive/ubuntu/rdiff-backup-development/+builds?build_text=&build_state=all and eventually the packages at https://launchpad.net/~rdiff-backup/+archive/ubuntu/rdiff-backup-development

**Comment by ericzolf** [comment]
Have you tried something like https://docs.travis-ci.com/user/encryption-keys/ ?
Basically, the idea would be:
- encrypt your SSH private key using travis (use ed25519 instead of RSA, makes the key smaller)
- add it to the .travis.yml as environment variable (the docs above explain how)
- in your script:
    - save the environment variable as an SSH key
    - tweak the `~/.ssh/config` to use this key for the ppa server (or define `GIT_SSH_COMMAND="ssh -i mysavedkey"` - see `man git`)
A bit convoluted but it should work IMHO...

**Comment by ericzolf** [comment]
Does this PR still make sense or shall it be been closed?

**Comment by ottok** [comment]
I will not be able to implement full PPA automation. In the long run a Salsa-CI mirror of upstream repository would be a better solution. Instead, document howto manually publish to the branch https://github.com/rdiff-backup/rdiff-backup/tree/ubuntu-ppa

**Comment by ottok** [comment]
Based on this I just pushed to https://launchpad.net/~rdiff-backup/+archive/ubuntu/rdiff-backup-development so the latest git master is available in the PPA for all supported Ubuntu releases:
image

I have not pushed anything to https://launchpad.net/~rdiff-backup/+archive/ubuntu/rdiff-backup-backports
Backports already have latest in 2.0.x series (2.0.5) - do you think there is any need to push there anything newer? Probably not, that PPA will be available just as an upgrade helper for anybody still on Ubuntu 18.04, and in a year or two we can delete the whole backports repository as obsolete.

File: docs/DEVELOP.adoc [hunk 1]
```diff
@@ -63,6 +63,7 @@ which acronym is not that important but here is a list of recommended ones (see
 * CHG: for a change requesting consideration when upgrading
 * DOC: for documentation aspects
 * WEB: anything regarding the website
+* DEV: relevant to developers
```
Comment (ericzolf) [inline docs/DEVELOP.adoc:66]: No, I want to remove it, I was getting sick (as main developer) to have to think if it's relevant to the changelog or not. Developers have directly a look at the `git log` and basta, the changelog is for users. We can have the discussion in #825 and align the docs there.

**Review by ericzolf — CHANGES_REQUESTED** [review]
Looks good to me, just one small change to rollback.

File: docs/DEVELOP.adoc (hunk 1 above)
Comment (ottok) [reply docs/DEVELOP.adoc]: ok, removed

**Review by ericzolf — APPROVED** [review]

**Comment by ottok** [comment]
@ericzolf Do you want to test the documentation supplied in this PR and see if you get a new build to show up at https://code.launchpad.net/~rdiff-backup/+recipe/rdiff-backup-git-master ?

**Comment by ericzolf** [comment]
> I have not pushed anything to https://launchpad.net/~rdiff-backup/+archive/ubuntu/rdiff-backup-backports 
Backports already have latest in 2.0.x series (2.0.5) - do you think there is any need to push there anything newer? Probably not, that PPA will be available just as an upgrade helper for anybody still on Ubuntu 18.04, and in a year or two we can delete the whole backports repository as obsolete.

Sorry, I oversaw this comment: I'd ask on the mailing list, it's for the users to decide (and you if you're willing to put the necessary effort to maintain it longer/actively).

> @ericzolf Do you want to test the documentation supplied in this PR and see if you get a new build to show up at https://code.launchpad.net/~rdiff-backup/+recipe/rdiff-backup-git-master ?

Which documentation exactly? I tried simply:

```
$ make dist_deb
docker run --rm -i -v /home/ericl/Comp/rdiff-backup/..:/build/ -w /build/rdiff-backup rdiff-backup-dev:debian-sid debian/autobuild.sh
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
✔ docker.io/library/rdiff-backup-dev:debian-sid
Trying to pull docker.io/library/rdiff-backup-dev:debian-sid...
Error: initializing source docker://rdiff-backup-dev:debian-sid: reading manifest debian-sid in docker.io/library/rdiff-backup-dev: errors:
denied: requested access to the resource is denied
unauthorized: authentication required

make: *** [Makefile:56: dist_deb] Error 125
```

I tried multiple registries but none seems to know your container image:

```
  ▸ registry.fedoraproject.org/rdiff-backup-dev:debian-sid
    registry.access.redhat.com/rdiff-backup-dev:debian-sid
    docker.io/library/rdiff-backup-dev:debian-sid
    quay.io/rdiff-backup-dev:debian-sid
```

So, good point, I might have wanted to test before merging :shrug:

**Pull request merged**